### PR TITLE
auth v1

### PR DESCRIPTION
**Pull request by JeremiSavard**
What changes does this PR contain?

Added new package to verify that an access token is provided when calling a microservice. Also verify the validity of the token. 

UBI-309
Does this PR change the outward facing API?

Will have an impact of every microservice except microservice-authentication 

Additional comments and information

Only work if keycloak is properly deployed

@Ubivius/microservices

### CODE REVIEW

File: auth/authenticationMiddleware.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package authentication
```
Comment (ArcticWalrus) [inline auth/authenticationMiddleware.go]: File name is redundant. Should just be auth/middleware.go

File: auth/authenticationMiddleware.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	configURL := "http://localhost:8080/auth/realms/ubivius"
+	ctx = context.Background()
+	provider, err := oidc.NewProvider(ctx, configURL)
+	if err != nil {
+		log.Println("Auth panic")
+		panic(err)
+	}
+
+	clientID := "ubivius-client"
+	clientSecret := "7d109d2b-524f-4351-bfda-44ecad030eef"
```
Comment (ArcticWalrus) [inline auth/authenticationMiddleware.go]: I'm guessing these will be in env variables soon

**Review by ArcticWalrus — CHANGES_REQUESTED** [review]

File: auth/authenticationMiddleware.go (hunk 2 above)
Comment (JeremiSavard) [reply auth/authenticationMiddleware.go]: yeah, but we could let them be this way for now

File: auth/middleware.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	configURL := "http://localhost:8080/auth/realms/ubivius"
+	ctx = context.Background()
+	provider, err := oidc.NewProvider(ctx, configURL)
+	if err != nil {
+		log.Println("Auth panic")
+		panic(err)
+	}
+
+	clientID := "ubivius-client"
+	clientSecret := "7d109d2b-524f-4351-bfda-44ecad030eef"
```
Comment (MaximeStOnge) [inline auth/middleware.go:28]: Il va falloir surement ne pas mettre le secret en clair

**Review by MaximeStOnge — APPROVED** [review]

File: auth/middleware.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+		}
+
+		parts := strings.Split(rawAccessToken, " ")
+		if len(parts) != 2 {
+			log.Println("Missing token parts")
+			w.WriteHeader(400)
+			return
+		}
+		_, err := verifier.Verify(ctx, parts[1])
+		log.Println(err)
```
Comment (ArcticWalrus) [inline auth/middleware.go]: Don't log error before checking it on line 69

**Review by ArcticWalrus — APPROVED** [review]
One small change

**Pull request merged**